src/acpi.rs

```diff
@@ -419,3 +423,334 @@ pub fn process_msct() -> (
+                        let mem_desc = parse_nfit_spa_range_structure(entry);
+                        pmem_descriptors.push(mem_desc);
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_MEMORY_MAP => {
+                        let entry = iterator as *const ACPI_NFIT_MEMORY_MAP;
+                        log_nfit_region_mapping_structure(entry);
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_INTERLEAVE => {
+                        let entry = iterator as *const ACPI_NFIT_INTERLEAVE;
+                        log_nfit_interleave_structure(entry);
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_SMBIOS => {
+                        warn!("Unable to handle ACPI_NFIT_SMBIOS table")
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_CONTROL_REGION => {
+                        let entry = iterator as *const ACPI_NFIT_CONTROL_REGION;
+                        log_nfit_control_region_structure(entry);
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_DATA_REGION => {
+                        warn!("Unable to handle ACPI_NFIT_DATA_REGION table")
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_FLUSH_ADDRESS => {
+                        let entry = iterator as *const ACPI_NFIT_FLUSH_ADDRESS;
+                        log_nfit_flush_hint_structure(entry);
+                    }
+                    Enum_AcpiNfitType::ACPI_NFIT_TYPE_RESERVED => {
+                        let entry = iterator as *const ACPI_NFIT_PLATFORM_CAPABILITIES;
+                        log_nfit_platform_capabilities_structure(entry);
+                    }
+                    _ => unreachable!(),
+                }
+                iterator = iterator.add((*header).Length as usize);
+            }
+        } else {
+            debug!("ACPI NFIT Table not found.");
+        }
+        pmem_descriptors
+    }
+}
+
+#[repr(C, align(8))]
+struct Guid {
+    data1: u32,
+    data2: u16,
+    data3: u16,
+    index8: u8,
+    index9: u8,
+    index10: u8,
+    index11: u8,
+    index12: u8,
+    index13: u8,
+    index14: u8,
+    index15: u8,
+}
+
+impl fmt::Debug for Guid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{{ {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X}, {:#X} }}",
+            self.data1,
+            self.data2,
+            self.data3,
+            self.index8,
+            self.index9,
+            self.index10,
+            self.index11,
+            self.index12,
+            self.index13,
+            self.index14,
+            self.index15
+        )
+    }
+}
+
+impl From<&[u8; 16]> for Guid {
+    fn from(slice: &[u8; 16]) -> Guid {
+        let p: *const [u8; core::mem::size_of::<Guid>()] =
+            slice.as_ptr() as *const [u8; core::mem::size_of::<Guid>()];
+        unsafe { core::mem::transmute(*p) }
```

Comment:
I'd put a for documentation and sanity checking here
```rust
// Safety:
static_assertions::assert_eq_size!([u8; 16], Guid);
static_assertions::assert_eq_align!([u8; 16], Guid);
```

https://docs.rs/static_assertions/1.1.0/static_assertions/macro.assert_eq_size.html

Reply:
Done.